lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1191,6 +1206,56 @@ private Task claim(String taskId, boolean forceClaim)
     return task;
   }
 
+  private Task requestReview(String taskId, boolean force)
+      throws TaskNotFoundException, NotAuthorizedException, InvalidTaskStateException,
+          InvalidOwnerException {
+    String userId = taskanaEngine.getEngine().getCurrentUserContext().getUserid();
+    TaskImpl task;
+    try {
+      taskanaEngine.openConnection();
+      task = (TaskImpl) getTask(taskId);
+
+      TaskImpl oldTask = task.copy();
+      oldTask.setId(taskId);
+      oldTask.setExternalId(task.getExternalId());
+      oldTask.setAttachments(task.getAttachments());
+      oldTask.setSecondaryObjectReferences(task.getSecondaryObjectReferences());
+
+      if (!force && !task.getState().equals(TaskState.CLAIMED)) {
+        throw new InvalidTaskStateException(task.getId(), task.getState(), TaskState.CLAIMED);
+      }
+      if (!force && !task.getOwner().equals(userId)) {
+        throw new InvalidOwnerException(userId, task.getId());
+      }
+      if (task.getState().isEndState()) {
+        throw new InvalidTaskStateException(
+            task.getId(), task.getState(), TaskState.CLAIMED, TaskState.READY);
+      }
```

Comment:
merge this statement with the status check on line 1224? Might make it more readable
```
if (forced) {
  if (task is in endstate) {
    ERROR
  }
} else {
  if (task is not claimed) {
    ERROR
  }
}

Reply:
Just looked at the implementation of `claimTask` There we validate that the task is not in an end state first, then verify that the requirements when the operation is not forced. Please ajust accordingly.
```
if (task is in endstate) {
  ERROR
}
if (!forced and task is not claimed) {
  ERROR
}

Reply:
Ok :)